internal/controlplane/handlers_repositories.go

```diff
@@ -290,8 +290,13 @@ func (s *Server) DeleteRepositoryById(ctx context.Context,
 		return nil, err
 	}
 
-	// delete the repository
-	if err := s.store.DeleteRepository(ctx, repo.ID); err != nil {
+	provider, err := getProviderFromRequestOrDefault(ctx, s.store, in, repo.ProjectID)
+	if err != nil {
+		return nil, providerError(fmt.Errorf("provider error: %w", err))
```

Comment:
`providerError` will already wrap with `provider error: %w` if the error is not a "no rows found" error, so you should just be able to pass in `err` here.

Reply:
(Yeah, most of the other examples get this wrong, I'll send a PR to fix)

Reply:
Yeah, I copied other examples, should have checked. Done.